Civi/Share/CiviMRF/Client.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Civi\Share\CiviMRF;
+
+use Civi\Core\Service\AutoService;
+use CMRF\Core\Call;
+use CMRF\Exception\ApiCallFailedException;
+
+/**
+ * @service civi.share.civimrf
+ */
+class Client extends AutoService {
+
+  private Core $cmrfCore;
+
+  private string $connectorId;
+
+  public function __construct() {
+    $this->cmrfCore = new Core();
+  }
+
+  public function init($connectorId) {
+    $this->connectorId = $connectorId;
+    return $this;
+  }
+
+  public function getConnectorId() {
+    if (!isset($this->connectorId)) {
```

Comment:
If the connector ID is required it should be better injected in the constructor.

Reply:
As this is a service and the connector ID is not known until the client being injected in another service, this needs to be done after instantiation, but prior to actually using the service, see here: https://github.com/systopia/de.systopia.share/blob/d1481aef929e19fab6bf02102fe4984142ca20d1/Civi/Share/CiviMRF/ShareApi.php#L21-L27